replica/table.cc

```diff
@@ -623,6 +623,9 @@ compaction_group& table::compaction_group_for_token(dht::token token) const noex
                                                 idx, _cg_manager->log2_compaction_groups(), _compaction_groups.size(), token));
     }
     auto& ret = *_compaction_groups[idx];
+    if (token == dht::minimum_token() || token == dht::maximum_token()) {
```

Comment:
dht::is_minimum and is_maximum should be used since only the token kind matters here
So tokens might match is_minimum but compare as not equal to minimum_token.
Maybd we should fix that in `token::operator<=>`, but in any case, the _is_minimum/maximum predicates are more efficient (and inline).